### PR TITLE
Add org.onionware.onionmedia

### PR DESCRIPTION
**Pull request by onionware-github**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- filesystem=host, device=all to enable the user to save converted files in any place
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.onionware.onionmedia.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2023 Jaden Nebel -->
+<component type="desktop">
+  <id>org.onionware.onionmedia</id>
+  <launchable type="desktop-id">org.onionware.onionmedia.desktop</launchable>
+  <metadata_license>CC-BY-3.0</metadata_license>
+  <project_license>AGPL-3.0-only</project_license>
+  <name>OnionMedia X</name>
+  <summary>Free Media Downloader and Converter - Cross Platform</summary>
+  <content_rating type="oars-1.0"/>
```
Comment (hfiguiere) [inline org.onionware.onionmedia.appdata.xml:10]:
```suggestion
  <content_rating type="oars-1.1"/>
```

File: org.onionware.onionmedia.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+command: start.sh
+
+modules:     
+  - name: Desktop
+    buildsystem: simple
+    build-commands:
+      - cp -r . /app
+    sources:
+      - type: archive
+        url: https://github.com/onionware-github/OnionMedia-X/releases/download/2.0.0.0/flatpak-resources.zip
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:14]: why doesn't this include the appstream file?

File: org.onionware.onionmedia.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        url: https://github.com/onionware-github/OnionMedia-X/releases/download/2.0.0.0/OnionMediaX_2.0.0.0-linux.X64.zip
+        sha256: 0271e87a6d5c5d7044a4eab821fc7a8a5ee7ab37708bef074a001ba3e6ba417e
+        dest-filename: publish.zip
+        
+    cleanup:
+      - '*.pdb'
+      - 'FluentAvalonia.ProgressRing.xml'
+      - 'YoutubeDLSharp.xml'
+        
+finish-args:
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:35]: `finish-arg` is usually found above `modules`.

File: org.onionware.onionmedia.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  - --filesystem=host
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:36]: this is a very wide filesystem permissions.

File: org.onionware.onionmedia.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  - --share=network
+  - --share=ipc
+  - --socket=session-bus
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:39]: this is not allowed you need to figure out which dbus names you need.

File: org.onionware.onionmedia.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        
+  - name: OnionMedia
+    buildsystem: simple
+    build-commands:
+      - cp -r . /app/bin
+      - install -D start.sh /app/bin/start.sh
+      - chmod -R +x /app/bin
+    sources:
+      - type: archive
+        url: https://github.com/onionware-github/OnionMedia-X/releases/download/2.0.0.0/OnionMediaX_2.0.0.0-linux.X64.zip
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:26]: how will this work with the aarch64 flatpak that is built?

File: org.onionware.onionmedia.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        sha256: 0271e87a6d5c5d7044a4eab821fc7a8a5ee7ab37708bef074a001ba3e6ba417e
+        dest-filename: publish.zip
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:28]:
```suggestion
```

File: org.onionware.onionmedia.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
+app-id: org.onionware.onionmedia
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:1]: `onionware.org` doesn't exists. The app-id doesn't meet the requirements.

File: org.onionware.onionmedia.yml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+      - type: archive
+        url: https://github.com/onionware-github/OnionMedia-X/releases/download/2.0.0.0/flatpak-resources.zip
+        sha256: 4b97e11f562db8fbc2be0babed349f477af0ae848faaff5383cbc2459bceaf53
+        dest-filename: flatpak-resources.zip
+        
+  - name: OnionMedia
+    buildsystem: simple
+    build-commands:
+      - cp -r . /app/bin
+      - install -D start.sh /app/bin/start.sh
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:22]: I don't see `start.sh` but if it is in the tarball it has already been copied.

File: org.onionware.onionmedia.yml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      - chmod -R +x /app/bin
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:23]: I don't think you need to do that. `cp` keep the permissions.

File: org.onionware.onionmedia.appdata.xml
Comment (hfiguiere) [inline org.onionware.onionmedia.appdata.xml:1]: this file is never installed.

File: org.onionware.onionmedia.yml (hunk 2 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:14]: I forgot it and will implement, thank you.

File: org.onionware.onionmedia.yml (hunk 4 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:36]: The user should be free to store converted/downloaded files everywhere.

File: org.onionware.onionmedia.yml (hunk 5 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:39]: I'll change it. 👍

File: org.onionware.onionmedia.yml (hunk 6 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:26]: Should be only available for x64 at the moment, I will add a flathub.json file. Can I close this PR and create a new one or can I edit this PR?

File: org.onionware.onionmedia.yml (hunk 8 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:1]: Do I have to buy a domain or is there another way to find a good app ID?

File: org.onionware.onionmedia.yml (hunk 9 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:22]: Without `install -D start.sh /app/bin/start.sh` it cannot be executed.

File: org.onionware.onionmedia.yml (hunk 10 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:23]: Files like ffmpeg and yt-dlp cannot be executed without `chmod -R +x /app/bin`

File: org.onionware.onionmedia.appdata.xml
Comment (onionware-github) [reply org.onionware.onionmedia.appdata.xml:1]: I will put it into the flatpak-resources archive.

File: org.onionware.onionmedia.yml (hunk 8 above)
Comment (hfiguiere) [reply org.onionware.onionmedia.yml:1]: See https://docs.flatpak.org/en/latest/conventions.html#application-ids how to use github.

File: org.onionware.onionmedia.yml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: start.sh
+  
+finish-args:
+  - --filesystem=host
+  - --share=network
+  - --share=ipc
+  - --talk-name=org.freedesktop.Notifications
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:11]: are you sure this is necessary?

File: org.onionware.onionmedia.yml (hunk 9 above)
Comment (hfiguiere) [reply org.onionware.onionmedia.yml:30]: yeah but then line 31 you do chmod on everything

File: org.onionware.onionmedia.yml (hunk 6 above)
Comment (hfiguiere) [reply org.onionware.onionmedia.yml:34]: why do you want to close this?

Comment (onionware-github) [reply org.onionware.onionmedia.yml:34]: How can I add files to a pull request? I added a flathub.json file to my fork, but it doesn't appear here.

File: org.onionware.onionmedia.yml (hunk 8 above)
Comment (onionware-github) [reply org.onionware.onionmedia.yml:1]: Ok, so I can use io.github.onionware.onionmedia?

Comment (hfiguiere) [reply org.onionware.onionmedia.yml:1]: `io.github.onionware_github.onionmedia`

File: org.onionware.onionmedia.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
+app-id: io.github.onionware_github.onionmedia
```
Comment (hfiguiere) [inline org.onionware.onionmedia.yml:1]: the manifest file name must match

File: org.onionware.onionmedia.appdata.xml
Comment (hfiguiere) [reply org.onionware.onionmedia.appdata.xml:1]: then it should be removed here.

Comment (onionware-github) [reply org.onionware.onionmedia.appdata.xml:1]: Will the information from appdata appear on Flathub when there is no appdata file in this repo?

Comment (hfiguiere) [reply org.onionware.onionmedia.appdata.xml:1]: do you think I would tell you to remove it if it was needed?

Comment (onionware-github) [reply org.onionware.onionmedia.appdata.xml:1]: Ok sorry you're right xD, I removed it

**Comment by barthalion** [comment]
bot, build io.github.onionware_github.onionmedia

**Comment by barthalion** [comment]
```
~ 5.4s ❱ flatpak run io.github.onionware_github.onionmedia
/app/bin/start.sh: line 1: ﻿#!/bin/bash: No such file or directory
chmod: changing permissions of '.': Read-only file system
chmod: changing permissions of './OnionMedia X': Read-only file system
chmod: changing permissions of './libHarfBuzzSharp.so': Read-only file system
chmod: changing permissions of './libSkiaSharp.so': Read-only file system
chmod: changing permissions of './start.sh': Read-only file system
chmod: changing permissions of './Data': Read-only file system
```

This seems wrong.

**Comment by onionware-github** [comment]
> ```
> ~ 5.4s ❱ flatpak run io.github.onionware_github.onionmedia
> /app/bin/start.sh: line 1: ﻿#!/bin/bash: No such file or directory
> chmod: changing permissions of '.': Read-only file system
> chmod: changing permissions of './OnionMedia X': Read-only file system
> chmod: changing permissions of './libHarfBuzzSharp.so': Read-only file system
> chmod: changing permissions of './libSkiaSharp.so': Read-only file system
> chmod: changing permissions of './start.sh': Read-only file system
> chmod: changing permissions of './Data': Read-only file system
> ```
> 
> This seems wrong.

I know, but it works

**Comment by barthalion** [comment]
And it's also a poor excuse. There's no reason to recursively add the executable bit to every file like [here](https://github.com/onionware-github/OnionMedia-X/blob/main/OnionMedia.Avalonia/Platforms/Linux/Output/start.sh#L3) in the first place, and it's not going to do anything on Flatpak as /app is read-only.

**Comment by hfiguiere** [comment]
It's also wrong outside flatpak since that comes from upstream

**Comment by onionware-github** [comment]
> And it's also a poor excuse. There's no reason to recursively add the executable bit to every file like [here](https://github.com/onionware-github/OnionMedia-X/blob/main/OnionMedia.Avalonia/Platforms/Linux/Output/start.sh#L3) in the first place, and it's not going to do anything on Flatpak as /app is read-only.

I removed it from the manifest 👍

**Comment by barthalion** [comment]
bot, build io.github.onionware_github.onionmedia

**Pull request closed without merging**